### PR TITLE
Support authority callbacks

### PR DESCRIPTION
**Pull request by Vatyx**
Related engine PR: https://github.com/improbableio/UnrealEngine/pull/45

3 callbacks are based on the previous discussion we've had about it.

Documentation will be added.

### CODE REVIEW

File: SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp [hunk 1]
```diff
@@ -203,11 +203,19 @@ void USpatialReceiver::HandleActorAuthority(Worker_AuthorityChangeOp& Op)
 					{
 						Actor->RemoteRole = ROLE_SimulatedProxy;
 					}
+
+					Actor->OnAuthorityGained();
+				}
+				else if (Op.authority == WORKER_AUTHORITY_AUTHORITY_LOSS_IMMINENT)
```
Comment (m-samiec) [inline SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp]: Perhaps this should be a switch statement now.

**Comment by danielimprobable** [comment]
Will you add the docs as part of this PR?

File: SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp (hunk 1 above)
Comment (danielimprobable) [reply SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp]: Will there ever be more than the three callbacks? I think this is probably fine

**Comment by Vatyx** [comment]
Yes, I good docs boy.

File: SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp (hunk 1 above)
Comment (Vatyx) [reply SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp]: I think this is fine as well.

Comment (m-samiec) [reply SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp]: This day will live in infamy.

**Comment by girayimprobable** [comment]
LGTM, ping us again when docs are ready please.

File: SpatialGDK/Documentation/content/authority.md [hunk 2]
```diff
@@ -102,3 +102,45 @@ Actor on **non-owning client-worker**:
 * `RemoteRole = ROLE_Authority`
 
 A `PlayerController` possessing different `Pawn`s would change their role as expected; the newly-possessed Pawn will become an autonomous proxy on the client-worker while the older `Pawn` will become a simulated proxy.
+
+## Authority Callbacks
+
+Due to authority now being dynamic in the GDK, we've added events that can trigger behavior when authority changes. These events will only ever trigger on server-workers.
```
Comment (joshuahuburn) [inline SpatialGDK/Documentation/content/authority.md]: Is the 'now' part necessary?

Comment (Vatyx) [reply SpatialGDK/Documentation/content/authority.md]: Removed.

**Pull request merged**